script.reddit.reader/resources/lib/slideshow.py

```diff
@@ -0,0 +1,1451 @@
+                self.image_label.setLabel(desc_and_image[0])
+                self.image_label.setVisible(True)
+                
+                self.txt_background.setImage('srr_dlg-bg.png')
+                self.txt_background.setVisible(True)  
+            else:   #don't show text and text_background if no text 
+                self.image_label.setVisible(False)
+                self.txt_background.setVisible(False)
+                
+        image_control.setImage(desc_and_image[1])
+        image_control.setPosition(ctl_x, 0)
+        image_control.setWidth(ctl_width)  
+        image_control.setHeight(ctl_height)
+        image_control.setAnimations(slide_animation)
+        # show the image
+        image_control.setVisible(True)
+
+
+class TableDropScreensaver(ScreensaverBase):
+
+    MODE = 'TableDrop'
+    BACKGROUND_IMAGE = 'srr_blackbg.jpg'
+    IMAGE_CONTROL_COUNT = 20
+    FAST_IMAGE_COUNT = 0
+    NEXT_IMAGE_TIME = 1500
+    MIN_WIDTH = 500
+    MAX_WIDTH = 700
+
+    def load_settings(self):
+        self.NEXT_IMAGE_TIME = 1500 #int(addon.getSetting('tabledrop_wait'))
+
+    def process_image(self, image_control, image_url):
+        ROTATE_ANIMATION = (
+            'effect=rotate start=0 end=%d center=auto time=%d '
+            'delay=0 tween=circle condition=true'
+        )
+        DROP_ANIMATION = (
+            'effect=zoom start=%d end=100 center=auto time=%d '
+            'delay=0 tween=circle condition=true'
+        )
+        FADE_ANIMATION = (
+            'effect=fade start=0 end=100 time=200 '
+            'condition=true'
+        )
+        # hide the image
+        image_control.setVisible(False)
+        image_control.setImage('')
+        # re-stack it (to be on top)
+        self.xbmc_window.removeControl(image_control)
+        self.xbmc_window.addControl(image_control)
+        # calculate all parameters and properties
+        width = random.randint(self.MIN_WIDTH, self.MAX_WIDTH)
+        height = int(width / self.image_aspect_ratio)
+        x_position = random.randint(0, 1280 - width)
+        y_position = random.randint(0, 720 - height)
+        drop_height = random.randint(400, 800)
+        drop_duration = drop_height * 1.5
+        rotation_degrees = random.uniform(-20, 20)
+        rotation_duration = drop_duration
+        animations = [
+            ('conditional', FADE_ANIMATION),
+            ('conditional',
+             ROTATE_ANIMATION % (rotation_degrees, rotation_duration)),
+            ('conditional',
+             DROP_ANIMATION % (drop_height, drop_duration)),
+        ]
+        # set all parameters and properties
+        image_control.setImage(image_url[1])
+        image_control.setPosition(x_position, y_position)
+        image_control.setWidth(width)
+        image_control.setHeight(height)
+        image_control.setAnimations(animations)
+        # show the image
+        image_control.setVisible(True)
+
+
+class StarWarsScreensaver(ScreensaverBase):
```

Comment:
A Star Wars screensaver inside a reddit script? :)
How is that connected? or would it be possible to move this to a separate screensaver add-on?